### PR TITLE
[RLlib] Cleanup examples folder 07: Translate `custom_experiment`, `custom_logger`, `custom_progress_reporter` to new API stack.

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup examples folder 07: Translate `custom_experiment`, `custom_logger`, `custom_progress_reporter` to new API stack.

Plus:
* Fix a bug in Algorithm. We used to NOT update the weights of the local EnvRunner upon Algorithm.restore (when on new API stack). Yes, this was done eventually, but the weights immediately after the `restore()` call on the local EnvRunner (and all remote EnvRunners) are wrong.
* Enhance example script running utility function.
* Add missing docstrings to more example scripts.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/evaluation/evaluation_parallel_to_training.py [hunk 1]
```diff
@@ -1,6 +1,72 @@
...
+|-----------------------------+------------+-----------------+--------+
+| PPO_CartPole-v1_1377a_00000 | TERMINATED | 127.0.0.1:73330 |     25 |
++-----------------------------+------------+-----------------+--------+
++------------------+--------+----------+--------------------+
+|   total time (s) |     ts |   reward |   episode_len_mean |
+|------------------+--------+----------+--------------------|
+|          71.7485 | 100000 |   476.51 |             476.51 |
++------------------+--------+----------+--------------------+
+
+When running without parallel evaluation (`--evaluation-not-parallel-to-training` flag),
```
Comment (simonsays1980) [inline rllib/examples/evaluation/evaluation_parallel_to_training.py:56]: Yeah this is really cool! Parallel to training

File: rllib/examples/ray_tune/custom_progress_reporter.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+)
+
+
+if __name__ == "__main__":
+    # Force Tuner to use old progress output as the new one silently ignores our custom
+    # `CLIReporter`.
+    # TODO (sven): Find out why we require this hack.
+    import os
+
+    os.environ["RAY_AIR_NEW_OUTPUT"] = "0"
```
Comment (simonsays1980) [inline rllib/examples/ray_tune/custom_progress_reporter.py:84]: I guess it would help users, if we describe in one line why we set this env variable.

**Review by simonsays1980 — APPROVED** [review]
LGTM. ANother bunch of awesome examples!

File: rllib/examples/ray_tune/custom_progress_reporter.py (hunk 2 above)
Comment (sven1977) [reply rllib/examples/ray_tune/custom_progress_reporter.py:84]: We do this in the line above:
```
# Force Tuner to use old progress output as the new one silently ignores our custom
# `CLIReporter`.
```

**Pull request merged**